server/chat-plugins/chatlog.ts

```diff
@@ -1019,8 +1021,48 @@ export class RipgrepLogSearcher extends Searcher {
 	}
 }
 
-export const LogSearcher: Searcher = new (Config.chatlogreader === 'ripgrep' ? RipgrepLogSearcher : FSLogSearcher)();
+export class DatabaseLogSearcher extends Searcher {
+	database: PostgresDatabase;
+	constructor() {
+		super();
+		this.usePM = false;
+		this.database = new PostgresDatabase();
+	}
+	async findBattleLog(tier: ID, number: number) {
+		const SQL = require('sql-template-strings');
+		const results = await this.database.query(
+			SQL`SELECT log FROM battle_logs WHERE roomid = ${number}`
+		);
+		if (!results || !results.length) return null;
+		return results[0].log;
+	}
+	async getSharedBattles(userids: string[]) {
+		const SQL = require('sql-template-strings');
+		const query = SQL`SELECT roomid, format FROM battle_logs WHERE `;
+		query.append(SQL`(p1id = ${userids[0]} AND p2id = ${userids[1]})`);
+		query.append(` OR `);
+		query.append(SQL`(p1id = ${userids[1]} AND p2id = ${userids[0]})`);
```

Comment:
Missed discord comment: you have to have an index on these or it will be too slow